android_configure.py

```diff
@@ -61,6 +61,9 @@ def patch_android():
 elif platform.system() == "Linux":
     host_os = "linux"
     toolchain_path = android_ndk_path + "/toolchains/llvm/prebuilt/linux-x86_64"
+    if platform.machine() in {"x86_64", "x86"}:
+        os.environ['CC_host'] = os.popen("which gcc").read().strip()
+        os.environ['CCX_host'] = os.popen("which g++").read().strip()
```

Comment:
Doesn't python provide a utility for this? Rather than spawning a shell process?

Reply:
I ran ./android-configure on an x86_64 host, and in the generated Makefile, CC.host and CXX.host are set to use the Android NDK. However, dynamic libraries compiled with the NDK cannot be used on an x86 host. Therefore, it's necessary to modify the Python code to check the host architecture and generate a toolchain that matches the x86 architecture.

Reply:
I meant, there's no need to spawn a sub process, python has a builtin for finding an executable path IIRC